db/cache_db.go

```diff
@@ -37,63 +41,77 @@ func (db *CacheDB) Get(key []byte) []byte {
 	return db.parent.Get(key)
 }
 
+// Set sets the value for the given key in the cache. It does not write the value to the
+// underlying database. Set does not incur a performance penalty compared to SetNoLock.
 func (db *CacheDB) Set(key []byte, value []byte) {
 	db.mtx.Lock()
 	defer db.mtx.Unlock()
 
 	db.SetNoLock(key, value)
 }
 
+// SetSync sets the value for the given key in the cache. It does not write the value to the
+// underlying database. SetSync does not incur a performance penalty.
 func (db *CacheDB) SetSync(key []byte, value []byte) {
 	db.mtx.Lock()
 	defer db.mtx.Unlock()
 
 	db.SetNoLock(key, value)
 }
 
-// NOTE: Implements atomicSetDeleter
+// SetNoLock sets the value for the given key in the database. It does not write the value to the
+// underlying database. SetNoLock implements atomicSetDeleter.
 func (db *CacheDB) SetNoLock(key []byte, value []byte) {
 	db.cache[string(key)] = cDBValue{value, false}
 }
 
+// Delete deletes the value for the given key. It does not delete the value from the underlying
+// database. Delete does not incur a performance penalty compared to DeleteNoLock.
 func (db *CacheDB) Delete(key []byte) {
 	db.mtx.Lock()
 	defer db.mtx.Unlock()
 
-	db.cache[string(key)] = cDBValue{nil, true}
+	db.DeleteNoLock(key)
 }
 
+// DeleteSync deletes the value for the given key. It does not delete the value from the
+// underlying database. Delete does not incur a performance penalty.
 func (db *CacheDB) DeleteSync(key []byte) {
 	db.mtx.Lock()
 	defer db.mtx.Unlock()
 
-	db.cache[string(key)] = cDBValue{nil, true}
+	db.DeleteNoLock(key)
 }
 
-// NOTE: Implements atomicSetDeleter
+// DeleteNoLock deletes the value for the given key. It does not delete the value from the
+// underlying database. DeleteNoLock implements atomicSetDeleter.
 func (db *CacheDB) DeleteNoLock(key []byte) {
 	db.cache[string(key)] = cDBValue{nil, true}
 }
 
+// Close closes the underlying database by deallocating the handle. Any attempts to use the DB
+// after close will result in a panic.
```

Comment:
This probably isn't true, maybe we should clarify that the db should not be used after Close. But it won't panic in this case I don't thiknk ...

Reply:
That's right. It depends on the backing database.